src/Raven.Client/Util/NewLineInsensitiveStringComparer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Raven.Client.Util
+{
+    internal sealed class NewLineInsensitiveStringComparer : IEqualityComparer<string>
```

Comment:
I don't like this. This allocates multiple times during the equality process.
We are better off just normalizing the values once in `Compare`

Reply:
Simplified